src/NotesWriterSSC.cpp

```diff
@@ -453,7 +453,8 @@ static RString GetSSCNoteData( const Song &song, const Steps &in, bool bSavingCa
 			npsPerMeasureStrings.push_back(serialize(npsPerMeasure, ",", 3));
 		}
 		
-		lines.push_back( ssprintf( "#NPSPERMEASURE:%s;", join("|",npsPerMeasureStrings).c_str() ) );
+		RString npsPerMeasure = "#NPSPERMEASURE:" + join("|", npsPerMeasureStrings) + ";";
```

Comment:
if these strings are huge, we can save a lot more string copying/moving by avoiding the temporary strings and directly appending to `lines`

Reply:
e.g.
```
lines.push_back("#NPSPERMEASURE");
for (RString s : npsPerMeasureStrings) {
  if (not first) {
    lines.push_back("|");
  }
  lines.push_back(s);
}
lines.push_back(";")
```

Reply:
I don't think we want to push to `lines` since that'd add new lines between each thing, although I agree with the premise. We could instead create a new vector and then join it at the end to push back though.

```cpp
std::vector<RString> npsParts;
npsParts.push_back("#NPSPERMEASURE");
for (RString s : npsPerMeasureStrings) {
  // > 1 since we start with a push_back
  if (npsParts.size() > 1) {
    npsParts.push_back("|");
  }
  npsParts.push_back(s);
}
npsParts.push_back(";")

lines.push_back(join("", npsParts));
```

Similar thing below.

As a side not while we're here, can we also add + expose `MAXNPS` as a computer value?